server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/LocalResources.java

```diff
@@ -130,6 +134,9 @@ private boolean isLocalAlias(MailAddress mailAddress) throws UsersRepositoryExce
             .asStream()
             .map(mapping -> mapping.asMailAddress()
                 .orElseThrow(() -> new IllegalStateException(String.format("Can not compute address for mapping %s", mapping.asString()))))
-            .anyMatch(Throwing.predicate(this::isLocaluser).sneakyThrow());
+            .filter(Throwing.predicate(this::isLocaluser).sneakyThrow())
+            .findFirst()
+            .map(FunctionalUtils.toFunction(ownerAddress -> LOGGER.debug("{} belongs to {} local user", mailAddress.asString(), ownerAddress.asString())))
```

Comment:
you don't really map the value, do you?
do you want to use `peek`?

Reply:
I could do that if I move it one line up, yes.